inc/saimacsec.h

```diff
@@ -0,0 +1,1478 @@
+     * and for egress.
+     *
+     * @type sai_u8_list_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_SECTAG_OFFSETS_SUPPORTED,
+
+    /**
+     * @brief Macsec MTU capability on system side (not including Macsec overhead).
+     *
+     * @type sai_uint16_t
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_SYSTEM_SIDE_MTU,
+
+    /**
+     * @brief Warm boot is supported for all saimacsec objects.
+     *
+     * @type bool
+     * @flags READ_ONLY
+     */
+    SAI_MACSEC_ATTR_WARM_BOOT_SUPPORTED,
+
+    /**
+     * @brief When false disables creation of saimacsec objects during Phy warm-boot.
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_ATTR_WARM_BOOT_ENABLE,
+
+    /**
+     * @brief TPID value used to identify packet C-tag.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0x8100
+     */
+    SAI_MACSEC_ATTR_CTAG_TPID,
+
+    /**
+     * @brief TPID value used to identify packet S-tag.
+     *
+     * @type sai_uint16_t
+     * @flags CREATE_AND_SET
+     * @default 0x88A8
+     */
+    SAI_MACSEC_ATTR_STAG_TPID,
+
+    /**
+     * @brief Global setting of read-clear or read-only for statistics read.
+     * The mode parameter for get_macsec_<foo>_stats_ext should match this.
+     *
+     * @type sai_stats_mode_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_MACSEC_ATTR_STATS_MODE,
+
+    /**
+     * @brief Enables physical bypass of MACsec module. Packets can physically
+     * bypass MACsec module only when there is no macsec_port object
+     * associated with Phy line-side port(s).
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_MACSEC_ATTR_PHYSICAL_BYPASS_ENABLE,
+
+    /**
+     * @brief List of MACsec ports
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_PORT
+     */
+    SAI_MACSEC_ATTR_PORT_LIST,
+
+    /**
+     * @brief List of all MACsec flow objects
+     *
+     * @type sai_object_list_t
+     * @flags READ_ONLY
+     * @objects SAI_OBJECT_TYPE_FLOW
+     */
+    SAI_MACSEC_ATTR_FLOW_LIST,
+
+    /**
+     * @brief End of MACsec attributes
+     */
+    SAI_MACSEC_ATTR_END,
+} sai_macsec_attr_t;
+
+
+/**
+ * @brief Attribute Id for sai_macsec_port
+ *
+ * @flags Contains flags
+ */
+typedef enum _sai_macsec_port_attr_t
+{
+    /**
+     * @brief Start of MACsec Port attributes
+     */
+    SAI_MACSEC_PORT_ATTR_START,
+
+    /**
+     * @brief MACSEC direction
+     *
+     * @type sai_macsec_direction_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_MACSEC_PORT_ATTR_MACSEC_DIRECTION = SAI_MACSEC_PORT_ATTR_START,
+
+    /**
+     * @brief Phy line-side Port Id associated with this MACsec port
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_PORT
+     */
+    SAI_MACSEC_PORT_ATTR_PHY_LINE_PORT_ID,
```

Comment:
SAI_MAC_PORT_ATTR_PORT_ID so that we can bind mac sec to the switch asic port.

Reply:
SAI_MACSEC_PORT_ATTR_PORT_ID